### PR TITLE
Perf improvements

### PR DESCRIPTION
**Pull request by denar90**
I noticed that using regular 3g network in Chrome, there are present some place for improvements.

**Changes**:
 - scripts now loaded with `defer`
 - non critical css was moved into the body
 - vimeo iframe is loading on `onLoad` event (perfect way to load it when user scroll to place where it's shown)

**Benefits**
---

**First paint before**
brunch - ultra-fast html5 build tool 2016-10-25 22-56-46

**First paint after**
brunch - ultra-fast html5 build tool 2016-10-25 22-57-43

Using `pwmetrics` I got this:

1 bash 2016-10-25 22-50-51

1 bash 2016-10-25 23-14-09

P.S. It looks like none run build during some period of time, hence doc in source ahead of docs in root..

### CODE REVIEW

**Comment by shvaikalesh** [comment]
Hey Artem, thanks for the awesome effort! I have a few minor notes on the code.

File: source/app/init.js [hunk 1]
```diff
@@ -55,6 +55,44 @@ function setTitle(contentClass) {
   document.title = heading.textContent + ' — Brunch';
 }
 
+function deferIframe() {
+  var deferIframes = document.getElementsByClassName('defer-iframe');
```
Comment (shvaikalesh) [inline source/app/init.js]: We don't need live collection, do we? I think `document.querySelectorAll('.defer-iframe')` will be a little better.

File: source/app/init.js [hunk 2]
```diff
@@ -55,6 +55,44 @@ function setTitle(contentClass) {
...
+  var deferIframesNumber = deferIframes.length;
+
+  if (deferIframesNumber) {
+    for (var i = 0; i < deferIframesNumber; i++) {
```
Comment (shvaikalesh) [inline source/app/init.js]: I think we can drop the `.length` check and do `for/of` loop here.

File: source/app/init.js [hunk 3]
```diff
@@ -55,6 +55,44 @@ function setTitle(contentClass) {
...
+      var iframe = document.createElement('iframe');
+      var frameborder = document.createAttribute('frameborder');
+      var webkitallowfullscreen = document.createAttribute('webkitallowfullscreen');
+      var mozallowfullscreen = document.createAttribute('mozallowfullscreen');
+      var allowfullscreen = document.createAttribute('allowfullscreen');
+
+      frameborder.value = '0';
+
+      iframe.setAttributeNode(frameborder);
```
Comment (shvaikalesh) [inline source/app/init.js]: `iframe.setAttribute('allowfullscreen', '')` or even `iframe.allowFullscreen = true` + shim

Comment (shvaikalesh) [inline source/app/init.js]: brunch.io visitors are mainly developers with evergreen or even canary browsers, so prefixes are extra.

File: source/app/init.js [hunk 4]
```diff
@@ -55,6 +55,44 @@ function setTitle(contentClass) {
...
+      iframe.setAttributeNode(webkitallowfullscreen);
+      iframe.setAttributeNode(mozallowfullscreen);
+      iframe.setAttributeNode(allowfullscreen);
+
+      iframe = setDataValuesToIframe(iframe, Object.assign({}, deferIframes[i].dataset));
```
Comment (shvaikalesh) [inline source/app/init.js]: `Object.assign(iframe, deferIframe.dataset)` and shim for Safari to support `.style` setter.

**Comment by denar90** [comment]
@shvaikalesh @paulmillr do I have to add changes due to comments in another PR?

**Comment by paulmillr** [comment]
yes

**Pull request merged**